presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationSmokeTest.java

```diff
@@ -134,6 +139,27 @@ private void insertData(byte[] data)
         }
     }
 
+    private void createMessagesWithHeader(String topicName)
+    {
+        try (KafkaProducer<byte[], byte[]> producer = createProducer()) {
+            // Message without headers
+            ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(topicName, null, "{}".getBytes(UTF_8));
+            producer.send(record);
+            // Message with simple header
+            record = new ProducerRecord<>(topicName, null, "{}".getBytes(UTF_8));
+            record.headers()
+                    .add("notfoo", "some value".getBytes(UTF_8));
+            producer.send(record);
+            // Message with multiple same key headers
+            record = new ProducerRecord<>(topicName, null, "{}".getBytes(UTF_8));
+            record.headers()
+                    .add("foo", "bar".getBytes(UTF_8))
+                    .add("foo", null)
```

Comment:
thanks for taking care of null values
can header key be null as well?

Reply:
No, at least the Kafka Java client throws an exception if you try to create a Header with null as key.
See: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/header/internals/RecordHeader.java#L32